### PR TITLE
Timeseries Panel: render threshold at either top or bottom of graph when it is out…

### LINKED ISSUE #41631: Alerting: Fix out of bounds threshold handle
If you set a threshold that is higher or lower than the current data, the thresholds handle is not visible.

### PR DESCRIPTION
**Pull request by gillesdemey**
**What this PR does / why we need it**:

This make the thresholds render at either the top or the bottom of a graph panel if the value is out of bounds. This avoids confusion when creating alerts with classic condition expressions.

https://user-images.githubusercontent.com/868844/143028365-03f12b73-4e87-4569-947a-a76df2173d9c.mov

**Which issue(s) this PR fixes**:

Fixes #41631

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/panel/timeseries/plugins/ThresholdControlsPlugin.tsx [hunk 1]
```diff
@@ -53,16 +53,26 @@ export const ThresholdControlsPlugin: React.FC<ThresholdControlsPluginProps> = (
...
+      const height = plot.bbox.height / window.devicePixelRatio;
+
+      // the threshold is below the visible series area
+      if (yPos < 0) {
+        handleYPos = 0;
+      }
+
+      // the threshold is above the visible series area
+      if (yPos > height) {
+        handleYPos = height;
```
Comment (dprokop) [inline public/app/plugins/panel/timeseries/plugins/ThresholdControlsPlugin.tsx:67]: Got the idea, but wonder if this won't create even more confusion with the threshold positioned against the wrong y-value. Maybe the handle should also have some kind of indicator that it's rendereder for a value that's outside the bound? I.e. dimmed handle plus indicator icon? I see unicons have these icons that would >possibly< work:

image
image

image


WDYT @peterholmberg ?

Comment (gillesdemey) [reply public/app/plugins/panel/timeseries/plugins/ThresholdControlsPlugin.tsx:67]: I was thinking about this as well, a visual cue to indicate it's out-of-bounds would help to make it more clear 👍

Comment (gillesdemey) [reply public/app/plugins/panel/timeseries/plugins/ThresholdControlsPlugin.tsx:67]: I tried a few different ones, so far I prefer the `angle-down` and `angle-up`. Screenshot 2021-11-15 at 18 34 51
Screenshot 2021-11-15 at 18 31 53
Screenshot 2021-11-15 at 18 30 11

Comment (gillesdemey) [reply public/app/plugins/panel/timeseries/plugins/ThresholdControlsPlugin.tsx]: I've updated the PR description with a video of the new behavior and the updated styles

**Review by peterholmberg — APPROVED** [review]
Awesome

**Pull request merged**